app/api/databaseInteraction/findURL.js

```diff
@@ -1,23 +1,27 @@
 var winston = require('winston');
+var MongoClient = require('mongodb').MongoClient;
 
 //	Search the database for the short_url or the original_url
 // depending on the key
-module.exports = function(db, key, paramUrl, callback) {
-	winston.log('info', 'Querying...');
-	// get the sites collection
-	var sites = db.collection('sites');
-	// set key for findOne function
-	var query = {};
-	query[key] = paramUrl;
-	// find document with url the user passed
-	sites.findOne(query, function(e, result) {
-		if (e) {
-			winston.log('error', e);
-		}	else if (result) {
-			callback(result);
-		}	else {
-			winston.log('info', 'Matching query does not exist');
-			callback(null);
-		}
+module.exports = function(dbURL, key, paramUrl) {
+	return new Promise(function(resolve, reject) {
+		MongoClient.connect(dbURL)
+		.then(function(db) {
+			winston.log('info', 'Querying...');
+			// get the sites collection
+			var sites = db.collection('sites');
+			// set key for findOne function
+			var query = {};
+			query[key] = paramUrl;
+
+			sites.findOne(query).then(function(value) {
+				db.close();
+				winston.log('info', 'Database Closed');
+				resolve(value);
+			});
+		})
+		.catch(function(reason) {
+			reject(reason);
+		});
```

Comment:
the `new Promise` is not needed here. `MongoClient.connect` returns a Promise! Just do:
```
return MongoClient.connect(dbURL).then(function(db) { 
...
});
```